prog/weaver/http.go

```diff
@@ -19,6 +19,8 @@ import (
 	weave "github.com/weaveworks/weave/router"
 )
 
+var allConnectionStates = []string{"established", "pending", "retrying", "failed", "connecting"}
```

Comment:
I agree with your thought; maybe we can do that as part of a more general re-org.